### PR TITLE
fix: update to new devnet naming and versioning system (0.18 Core)

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

In the latest Core 0.18, a devnet versioning system was added.  This changes the devnet genesis block, the user agent and when DGW is used.  

Source:  https://github.com/dashpay/dash/pull/4685

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/params/DevNetParams.java
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/params/DevNetParams.java:46]: `devnetVersionName` should not be static, each devnet has its own name

**Comment by HashEngineering** [comment]
Before this is finished, I should try to get the difficulty algo working again for devnets.

**Review by Syn-McJ — APPROVED** [review]
Looks good

**Comment by HashEngineering** [comment]
This PR needs to wait until the similar PR is merged in Dash Core.

**Comment by HashEngineering** [comment]
This change was not added back into Core 18.  Let us close it for now.

**Pull request closed without merging**
